### DISCUSSION TITLE
Disable es5 transpilation while developing

### DISCUSSION THREAD

**Discussion by jonluca** [vercel/next.js#37595] [Help]
Is there a way to disable all the polyfills, variable renaming, and transpilation while developing? 

I've found that chrome devtools sometime has issues with source maps, and that stepping over some async functions that get turned into generators lead to a lot of fluff in the stack traces. If we're developing on the latest version of Chrome, is there an option to be able to disable all non-required build optimizations?

All the previous workarounds no longer work, [as mentioned on StackOverflow]

**Comment by icyJoseph** [comment]
I know that with browserlist you can tell some engines to transpile the code differently in development environments, but, I think, since we are in the early days of SWC in Next.js (which might be the reason why the old workarounds don't work), we can't do this just yet. 

Normally you could control this through  a `.swcrc` file, but it looks like Next.js is not honouring that just now.

- https://swc.rs/docs/configuration/swcrc
- https://github.com/browserslist/browserslist#configuring-for-different-environments
- Somewhat related: https://github.com/vercel/next.js/issues/37142

**Reply by jonluca** [reply]
Thanks - I got it to a better state by using

```js
const isProd = process.env.NODE_ENV === "production";

const nextConfig = {
  webpack: (config, { isServer, dev }) => {
    if (!isServer && dev) {
      config.optimization.minimize = false;
      delete config.optimization.minimizer;
      config.target = ["web", "es2020"];
    }
    return config;
  }
};

if (!isProd) {
  nextConfig.experimental = {
    legacyBrowsers: false,
    browsersListForSwc: true,
  };
}

module.exports = nextConfig;
```

as my config, but it's still not great. Being able to configure all of this via swc will be nice.

Thanks!

**Reply by icyJoseph** [reply]
I wonder if you could use phases... Still same as your proposal though.

```js
/** @type {import('next/constants')} */
const { PHASE_DEVELOPMENT_SERVER } = require("next/constants");

/** @type {(phase: string) => import('next').NextConfig} */
const nextConfig = (phase) => {
  const experimental =
    phase === PHASE_DEVELOPMENT_SERVER
      ? {
          legacyBrowsers: false,
          browsersListForSwc: true,
        }
      : {};

  return {
    reactStrictMode: true,
    experimental,
    webpack: (config) => {
      if (phase === PHASE_DEVELOPMENT_SERVER) {
        config.optimization.minimize = false;
        delete config.optimization.minimizer;
        config.target = ["web", "es2020"];
      }

      return config;
    },
  };
};



module.exports = (phase) => {
  return nextConfig(phase);
};

module.exports = nextConfig;
```

**Reply by jonluca** [reply]
Nice, I didn't know about phases. Cleaner than my solution, thanks

I'm motivated to fix this because I think at some point something in my toolchain is causing sourcemaps to be incorrect. You can see in the video below, that when I try and set a breakpoint on a certain line, it gets set elsewhere, and never actually gets triggered (in either place) when the code is executed.

https://user-images.githubusercontent.com/13029040/174152891-29457f64-7be3-4b06-92b5-fbcf7fee366d.mov

I'm not good enough at understanding sourcemaps to understand whats going on, but if I had to guess there's something wonky in the web3js libraries that cause either webpack, babel/swc, or maybe even chrome dev tools to have a subtle bug. I know that in my other apps, where I see this same behavior, that turning off transpilation and just using all native browser features fixes it (for the most part).

Getting my debugger back would be nice. This unfortunately persists across code changes, cache cleans, browser restarts, etc. It's a "real" bug AFAICT
